coral-spark/src/test/java/com/linkedin/coral/spark/CoralSparkTest.java

```diff
@@ -354,11 +354,11 @@ public void testSchemaPromotionView() {
   @Test
   public void testUnionExtractUDF() {
     RelNode relNode = TestUtils.toRelNode("SELECT extract_union(foo) from union_table");
-    String targetSql = String.join("\n", "SELECT extract_union(foo)", "FROM default.union_table");
```

Comment:
Does this test still simulate the actual environment? In the test, `foo` is a uniontype column. In a real environment, `foo` would have been converted to a struct by the underlying reader.

Reply:
Yes, I think it simulates the real environment. To my understanding, the entire logic flow is: spark get a hive view -> spark to ask coral to translate the view -> spark then plan the execution based on the translated view -> the scan step in the plan will be delegated to datasoureV2 reader.

So at Coral step, the underlying reader hasn't come to play yet, coral will simply query HMS thus getting the uniontype back.